### PR TITLE
Refactor docker setup

### PR DESCRIPTION
**Pull request by gedoensmax**
- Removes poetry from the repository
- simplifies Dockerfile (partly at the expense of a bigger resulting container due to building in the final container)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
+	docker run --gpus all --rm -v $(shell pwd):/repo --workdir /repo --entrypoint "make" vpf-gpu-trt run_samples_without_docker
 
 run_samples_without_docker:
 	wget http://www.scikit-video.org/stable/_static/bikes.mp4
-
-	python ./samples/SampleDecode.py -g 0 -e ./bikes.mp4 -r ./tests/test.raw       
-	python ./samples/SampleDecodeSw.py -e ./bikes.mp4 -r ./tests/test.raw
-	python ./samples/SampleEncodeMultiThread.py 0 848 464 ./tests/test.raw 10
-	python ./samples/SampleMeasureVideoQuality.py -g 0 -i ./tests/test.raw -o ./tests/test.raw -w 848 -h 464   
+	python3 -m pip install -r ./samples/requirements.txt
```
Comment (theHamsta) [inline Makefile:20]: could the samples be properly integrated into the tests? and we get ride of the Makefile? Or too much effort?

File: docker/Dockerfile.tensorrt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+ARG GEN_PYTORCH_EXT=1
+# Build vpf
+RUN mkdir /src
+WORKDIR /src
+COPY src src
+COPY setup.py setup.py
+COPY CMakeLists.txt CMakeLists.txt
+RUN python3 -m pip install --no-cache-dir scikit-build \
+    && python3 -m pip install --no-cache-dir .
```
Comment (theHamsta) [inline docker/Dockerfile.tensorrt:29]: pip should automatically install scikit-build. if not pip/setuptools is too old.

File: samples/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+pycuda
+tqdm
+torchvision
+onnx
+tensorrt
```
Comment (theHamsta) [inline samples/requirements.txt:5]: could we just add this to the extras of setup.py? There would otherways to much sets of requirements.

Extras can be installed by naming them during pip install `pip install .[samples]` to install all dependencies of the "samples" extra in setup.py or `pip install .[samples-all]`/`pip install .[samples-pytorch]`

File: docker/Dockerfile.tensorrt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    wget \
+    build-essential \
+    ninja-build \
+    cmake \
+    git \
+    python3 \
+    python3-pip \
+    python-is-python3
+
+ARG GEN_PYTORCH_EXT=1
```
Comment (theHamsta) [inline docker/Dockerfile.tensorrt:21]: not needed anymore I assume?

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -23,26 +23,30 @@ jobs:
...
       
-    - name: Setup Poetry
-      run: poetry install --extras vpf-samples        
-
+    - name: Setup pip
+      run: |
+        python3 -m venv /tmp/venv_vpf
+        source /tmp/venv_vpf/bin/activate
+        python3 -mpip install -r ./samples/requirements.txt    
+
```
Comment (theHamsta) [inline .github/workflows/test.yml:31]: add
```
on:
  schedule:
    - cron: '0 0 * * *'  # every day at midnight
```
in addition to current triggers to run the job daily? poetry broke due to an external dependency update.

File: Makefile (hunk 1 above)
Comment (theHamsta) [reply Makefile:20]: @rnaskulwar actually created a list of tests that he ran manually. We could download the contents in the test files

File: docker/Dockerfile.tensorrt (hunk 4 above)
Comment (theHamsta) [reply docker/Dockerfile.tensorrt:21]: oh, it is still used.

Comment (theHamsta) [reply docker/Dockerfile.tensorrt:21]: maybe use the desired extras (in setup.py) as a docker arg? PIP_INSTALL_EXTRAS=samples

File: README.md [hunk 6]
```diff
@@ -78,7 +79,7 @@ After resolving those you should be able to run `make run_samples_without_docker
 ```pwsh
 # Indicate path to your FFMPEG installation (with subfolders `bin` with DLLs, `include`, `lib`)
 $env:SKBUILD_CONFIGURE_OPTIONS="-DTC_FFMPEG_ROOT=C:/path/to/your/ffmpeg/installation/ffmpeg/" 
-
+pip install scikit-build
```
Comment (theHamsta) [inline README.md:82]: should not be necessary

File: README.md [hunk 7]
```diff
@@ -101,12 +102,11 @@ are required)
 
 ```bash
 DOCKER_BUILDKIT=1 docker build \
-                --tag vpf-gpu-all \
-                -f docker/Dockerfile.gpu \
+                --tag vpf-gpu \
+                -f ocker/Dockerfile \
```
Comment (theHamsta) [inline README.md:106]: typo

File: README.md [hunk 8]
```diff
@@ -101,12 +102,11 @@ are required)
...
                 --build-arg GEN_PYTORCH_EXT=1 \
-                --build-arg GEN_OPENGL_EXT=1 \
                 .
-docker run -it --rm --gpus=all vpf-gpu-all
+docker run -it --rm --gpus=all vpf-gpu
```
Comment (theHamsta) [inline README.md:109]: Can we remove `. Check the [Wiki page](https://github.com/NVIDIA/VideoProcessingFramework/wiki/Building-from-source) on how to build from source.` from README

File: README.md (hunk 6 above)
Comment (theHamsta) [reply README.md:82]: if older versions of setuptools can't do this, then it probably better to error out

File: Makefile (hunk 1 above)
Comment (gedoensmax) [reply Makefile:20]: The make file annoys me as well but i was too lazy to search for all references of it. Sure we can make it obsolete.

File: docker/Dockerfile.tensorrt (hunk 2 above)
Comment (gedoensmax) [reply docker/Dockerfile.tensorrt:29]: Aaaah ok interesting never worked for me is this a very recent version that is needed ?

File: samples/requirements.txt (hunk 3 above)
Comment (gedoensmax) [reply samples/requirements.txt:5]: I like that idea ! I'll check it out.

File: docker/Dockerfile.tensorrt (hunk 2 above)
Comment (theHamsta) [reply docker/Dockerfile.tensorrt:29]: It think the problem is that the info is stored in our (cluttered) pyproject.toml which is only regarded by newer versions of setuptools
https://github.com/theHamsta/VideoProcessingFramework/blob/fdf418647dc1cf25cb1299b4066c5b3d5d1b26f9/pyproject.toml#L31-L39

File: README.md (hunk 8 above)
Comment (gedoensmax) [reply README.md:109]: done

**Pull request merged**